### PR TITLE
HOTFIX: fix failing test PlaintextAdminIntegrationTest.testElectPreferredLeaders

### PR DESCRIPTION
**Pull request by agam**
(broken by https://github.com/apache/kafka/pull/3909)

- Confirmed the test passes after this (minor) change.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by hachikuji** [comment]
ok to test

**Comment by abbccdda** [comment]
Why would this fix work? The exception message doesn't have a period ending.

**Comment by hachikuji** [comment]
@abbccdda Isn't that the point? The assertion now expects the period.

**Review by abbccdda — COMMENTED** [review]
Ok, seems working on local

**Comment by abbccdda** [comment]
@hachikuji I don't remember the version but yesterday when I look at it, there is no period in the end :)

**Comment by agam** [comment]
Actually, even this fix seems fragile, it should either check for some substring, or just refer to the _symbolic_ error code and get its message. Made the latter change, PTAL and let me know if that makes sense.

File: core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala [hunk 1]
```diff
@@ -1285,7 +1285,7 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
     var electResult = client.electLeaders(ElectionType.PREFERRED, Set(partition1).asJava)
     var exception = electResult.partitions.get.get(partition1).get
     assertEquals(classOf[ElectionNotNeededException], exception.getClass)
-    assertEquals("Leader election not needed for topic partition", exception.getMessage)
+    assertEquals(org.apache.kafka.common.protocol.Errors.ELECTION_NOT_NEEDED.message(), exception.getMessage)
```
Comment (hachikuji) [inline core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala]: Is the assertion useful in the first place? Verifying the exception type seems good enough

Comment (agam) [reply core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala]: True, we use `ApiError` for this, which pulls in the right message programmatically. Removed.

**Comment by hachikuji** [comment]
retest this please

**Comment by hachikuji** [comment]
ok to test

**Comment by ijuma** [comment]
One job passed, one failed due to flaky:

> kafka.server.DynamicBrokerReconfigurationTest.testUncleanLeaderElectionEnable

**Review by ijuma — APPROVED** [review]
It's debatable whether having the exception string is useful or not. We don't follow a consistent approach in Kafka. Since this is failing consistent in trunk, I'll go ahead and merge. LGTM.

**Comment by agam** [comment]
(confirmed that `testUncleanLeaderElectionEnable` passes locally for trunk)

**Pull request merged**